### DISCUSSION TITLE
Feedback

### DISCUSSION THREAD

**Discussion by govuk-design-system** [alphagov/govuk-design-system#4978] [Prioritised components and patterns]
Discussion regarding [this component in the GOV.UK Design System](https://design-system.service.gov.uk/components/feedback/).

**Comment by ashleyanderton** [comment]
Does anybody have any analytics data on typical engagement with this?

We tried an alternative on Explore careers where we made the survey banner sticky and adjusted the design to accommodate. Currently seeing around 2% of users providing feedback through the Yes / No links. Just wondering how that compares to non sticky.

Visible here: https://beta.nationalcareersservice.direct.gov.uk/job-profiles/police-officer

![image](https://user-images.githubusercontent.com/28305096/49526166-280d7e80-f8a7-11e8-8933-5fca841b0fc0.png)

Full write-up and further research can be found here:
https://docs.google.com/document/d/1GbEMdH578tnelCJQBy0o4Qn_JXRiMH2wmoiTboXmCys/edit?usp=sharing

**Comment by soupdragon99** [comment]
# Dropbox Paper audit

On 19th March 2019 the Design System team reviewed a Dropbox Paper document discussing the Feedback links pattern.

The aim was to reduce the number of places containing guidance and code by:

- migrating relevant, useful content into the Design System itself
- recording important research findings in the community backlog
- removing the original Dropbox Paper page

Below is a record of the outcome of that review.

If you need to, you can [see the original Dropbox Paper content in the internet archive](https://web.archive.org/web/20190319110338/https://paper.dropbox.com/doc/Feedback-links--AZkfLnH6xiLEHbjS~DU678hIAg-Zb9L2Ih5gSA0Bwg7pql9u).

## Review outcomes

Combine the document history discussion on Dropbox Paper with this issue and remove the original Dropbox Paper page.

# Feedback links

## Overview

Improve your service by helping users tell you what they think of it.

## When to use this pattern

All public beta or live services must have a feedback page so you can collect comments and [measure user satisfaction](https://www.gov.uk/service-manual/measuring-success/measuring-user-satisfaction).

## Examples
### GOV.UK information page feedback link

Here’s how the feedback links at the bottom of pages on GOV.UK work:

![](https://d2mxuefqeaa7sj.cloudfront.net/s_CCFBE5C4DB400C218D4E9CA7F499F6EEF29437BF8415BF782C839A659DC20BC9_1496919314193_feedback-link.gif)

### GOV.UK Service Manual page feedback link

On [Service Manual]() pages the feedback form is combined with a simple yes/no link for tracking user satisfaction:

![](https://d2mxuefqeaa7sj.cloudfront.net/s_CCFBE5C4DB400C218D4E9CA7F499F6EEF29437BF8415BF782C839A659DC20BC9_1496925487501_feedback-service-manual.gif)

### GOV.UK user survey

Every 50 page-loads we insert a banner: “Tell us what you think of GOV.UK – take the 3 minute survey". This takes you to this survey: https://www.surveymonkey.com/s/6HZFSVC

## Related patterns
- [+Feedback pages](https://paper.dropbox.com/doc/5IlCuNH7ZttnFvrKAJD9I) 
- [+Leave feedback](https://paper.dropbox.com/doc/vseI1yqypVT4JX6WxtEzv)

**Comment by terrysimpson99** [comment]
On one service we eliminated the feedback link and put the feedback questions directly on the feedback page. The feedback rate jumped from 2.5% (average over 6 months) to 45% (average over 1 month).

Can anyone think of a significant disadvantage to putting feedback questions directly on the confirmation page?

**Comment by NickColley** [comment]
@terrysimpson99 this pattern is used on many different types of pages, so I imagine putting them inline might not work so well for a generic use case.

> Can anyone think of a significant disadvantage to putting feedback questions directly on the confirmation page?

However, you might want to consider raising your findings on the [confirmation pages](https://github.com/alphagov/govuk-design-system-backlog/issues/40) pattern page, perhaps we could consider updating that pattern with what you've found?

**Comment by LydiaValtindOC** [comment]
Across HMRC Borders & Trade services, I've seen variations of links and forms to gather feedback:

https://docs.google.com/document/d/1GcajywDVL-0jZpTyWoxRuQJ1rnggp_1XiF4FJsPbl2c/edit?usp=sharing

**Comment by Ciandelle** [comment]
Merge alphagov/govuk-design-system-backlog#170 with this into a feedback link pattern rather than as two separate issues

**Comment by Ciandelle** [comment]
When a user wants assistance with an action they're attempting to complete on a page, there is a link which displays a form asking for information from the user which will be useful in providing them assistance.

## Screenshot of current live service:

![image](https://cloud.githubusercontent.com/assets/234825/22735065/fe7b5c4a-edef-11e6-8efd-4771ef5b1a9a.png)


## Slack Discussion

A slack [discussion](https://hmrcdigital.slack.com/archives/community-ux/p1486047900000074) on [#community-ux](https://hmrcdigital.slack.com/archives/community-ux) suggested that it would be useful to make the form more service specific.

The discussion also suggests that the answers provided to the question 'What were you doing?' were sometimes not resulting in useful responses from users.

**@philsherry commented on 8 Feb 2017**
While the fields for Name and Email are fine (although, `type="email"` would be a good addition to the latter…), the question fields should really be `textarea` rather than `input`.

It could do with behaving a bit nicer, too. With no JS, it just appears – that's fine, but this link is still sat in the page: `<a class="report-error__toggle js-visible" id="get-help-action" href="Action(parser=BodyParser(anyContent))">Is there anything wrong with this page?</a>`. That `href` is _nasty_. 


**@gavinwye commented on 27 Apr 2017**
If the service that's using this pattern isn't set up in Deskpro the Deskpro team don't know where the feedback is coming from. Maybe we could add a hidden field with the service name to this form to make it easier for the Deskpro users.

**@jennifer-hodgson commented on 6 Jul 2017**
On BTA, we're also having issues with this pattern. Currently our users are tending to use this pattern to report issues with their tax affairs, which is generating a number of rogue, misdirected Deskpro tickets. It appears the distinction between "webpage issues" and "tax issues" isn't clear enough in the current content.

I would suggest that we need to make clearer that this form is for reporting issues with the page and/or the digital service, and also perhaps direct users towards the correct place for reporting issues with their tax affairs.

**@stevenaproctor commented on 6 Jul 2017**
I agree with @jennifer-hodgson. People tend to ignore the "Don't include any personal or financial information. For example, your National Insurance or credit card numbers." that is at the top of the form.

A lot of services changed the link to "Is there anything wrong with this page?" to try and stop this kind of behaviour but I do not think this has had the effect people were hoping for. Plus, all the content on the form is still geared towards "get help".

**@gavinwye commented on 9 Oct 2017**
Code for use in your prototype

```
<div> 
                <br>      <br>        
                 <p>
                  <a href="#support" class="report-error" id="get-help-action" onclick="$('#support').toggle();return false;">Get help with this page</a></p>
                  <small class="hidden-help" id="support" style="display: none;">
                
                  <div id="report-error-partial-form" class="report-error__content">
                  <h3 class="heading-medium">Get help using this service</h3>
                  <p>
                     Don't include any personal or financial information. For example, your National Insurance or credit card numbers.
                  </p>
                  <form action="" method="" class="form--feedback" novalidate="novalidate">
                      <fieldset>
                      
                      <div class="form-group-compound">
                              <label for="report-name" class="form-label">Name</label>
                              <input id="report-name" class="form-control-3-4" name="report-name" type="text" data-rule-letterswithbasicpunc="true" data-rule-required="true" data-msg-required="Please provide your name." aria-required="true">
                          </div>
                          <div class="form-group-compound">
                              <label for="report-email" class="form-label">Email</label>
                              <input id="report-email" class="form-control-3-4" name="report-email" type="email" data-rule-required="true" data-rule-email="true" data-msg-required="Please provide your email address." aria-required="true">
                          </div>
                          <div class="form-group-compound">
                              <label class="form-label" for="text">What where you doing?</label>
                              <input class="form-control-3-4" id="text" name="report-action" type="text" data-rule-required="true" data-msg-required="Please enter details of what you were doing." aria-required="true">
                          </div>
                          <div class="form-group-compound">
                              <label for="report-error" class="form-label">What do you need help with?</label>
                              <input id="report-error" class="form-control-3-4" name="report-error" type="text" data-rule-required="true" data-msg-required="Please enter details of what went wrong." aria-required="true">
                          </div>
                         
                          <input type="hidden" name="isJavascript" value="true">
                          <input type="hidden" name="referrer" value="https://web-qa.tax.service.gov.uk/mdtp/registration">
                          <div class="form-field">
                              <button id="report-submit" class="button" type="submit" data-journey-click="">Send</button>
                          </div>
                      </fieldset>
                  </form>
                  </div></small></div>
```

**@stevenaproctor commented on 17 Oct 2017**
@gavinwye This is a GOV.UK pattern. It is the same as what is on GOV.UK but with name and email address added to it.

**@JamesPBoyle commented on 1 Feb 2018**
@olirose @JSJohal - this is the design system thread for the get help with this page component.

@stevenaproctor, Oli and Jas are working on Making Tax Digital for Business. They did some user research on the get help form late last year. We've had a few chats about ways to improve the form.

We've currently agreed a process where Oli and Jas will kick around ideas. They'll touch base with platform services to confirm if the ideas are technically feasible to implement. Then they'll look to do user testing. I'm sure they'd welcome the ideas you had for the form.

We are expecting a bit of a lead time on this work. Rather than specifically testing designs in unique UR sessions, we are going to test the new patterns if users get stuck during a journey.

**@jennifer-hodgson commented 18 days ago**
This, again, is GOV.UK - we could feed our findings about people supplying the wrong kind of information and any new findings from the new version @stevenaproctor and I worked on to them. Cannot seem to find an issue on their backlog, though...

**Comment by stevenjmesser** [comment]
On the [Check planning and housing data for England](https://publish.planning.data.gov.uk) service we’re going to experiment with removing the feedback link from the phase banner and include it at the end of a journey instead. We’ll report back with findings once we have some.

**Comment by joelanman** [comment]
@stevenjmesser the risk there is if for whatever reason someone can't get to the end of a journey, they can't give feedback?

**Comment by stevenjmesser** [comment]
> @stevenjmesser the risk there is if for whatever reason someone can't get to the end of a journey, they can't give feedback?

There's a support channel that runs in parallel with feedback links, @joelanman, and these two routes to providing feedback work in tandem.

**Comment by Moji-Idowu** [comment]
In our service we are looking at how to better balance collecting exit survey feedback with keeping sign-in easy for a service that users access multiple times a day (to create and manage GMRs.)

Previously, the sign-in option was prioritised on the screen, with the Feedback link at the bottom . We adjusted it so that when a user clicks Sign out (following GDS standards), they’re automatically taken to the feedback survey with this, they can’t immediately sign back in. (from this implementation we got thousands of constructive feedback, shooting up from 10)

So my question is, how are other services managing to collect feedback without frustrating users or making it harder for them to sign back in? Would love to hear examples or approaches that strike a good balance.

A good approach will be to add the "Sign-in" on the top right of the blue banner on the survey feedback page, so we can monitor and see if it works.
Another question that leads into this, is how long the timeout is regarding sign out? what is the standard and because it seems like a lot of responses included the automatic timeout and we are wondering if its happening too quickly in our service.

**Comment by owenatgov** [comment]
# We have published the Feedback component

The GOV.UK Design System published the [Feedback component](https://design-system.service.gov.uk/components/feedback/) on 27 August 2026 in [GOV.UK Frontend v6.5.0](https://github.com/alphagov/govuk-frontend/releases/tag/v6.5.0).

[Read about our design decisions for this component](https://github.com/alphagov/govuk-design-system/discussions/5588)

This component has been initially released in 'Trial' status. See [our separate discussion](https://github.com/alphagov/govuk-design-system/discussions/5619) on why Feedback has been released in trial and how you can help us move it into stable status. You can also read more about our [component lifecycle statuses](https://design-system.service.gov.uk/community/component-lifecycle-statuses/).
